modules/gateway/gateway.go

```diff
@@ -159,18 +144,26 @@ func New(addr string, persistDir string) (g *Gateway, err error) {
 	g.log.Println("INFO: gateway created, started logging")
 
 	// Forward the RPC port, if possible.
-	go g.forwardPort(port)
+	g.threads.Add()
+	go func() {
+		defer g.threads.Done()
+		g.forwardPort(port)
```

Comment:
In the host, `forwardPort` changes the state of the host, mostly to grab the port number. I thought the gateway did the same thing, but it seems like it doesn't.